llvm/lib/Target/Xtensa/XtensaISelLowering.cpp

```diff
@@ -90,6 +90,22 @@ XtensaTargetLowering::XtensaTargetLowering(const TargetMachine &TM,
   setOperationAction(ISD::BR_CC, MVT::i64, Expand);
   setOperationAction(ISD::BR_CC, MVT::f32, Expand);
 
+  // Used by legalize types to correctly generate the setcc result.
+  setOperationPromotedToType(ISD::SETCC, MVT::i1, MVT::i32);
+  setOperationPromotedToType(ISD::BR_CC, MVT::i1, MVT::i32);
```

Comment:
Looks redundant, this should be the default behavior.
Also, BR_CC case is repeated above.

Reply:
@s-barannikov , thank you very much for comments. I removed this code and add more tests for branches.